src/ActionKit/CsrfToken.php

```diff
@@ -0,0 +1,61 @@
+<?php
+namespace ActionKit;
+use Exception;
+
+class CsrfToken {
+    
+    protected $timeout = 300;
+
+    public function __construct($timeout=300){
+        $this->timeout = $timeout;
+    }
+
+    public function randomString($len = 10) {
+        // Characters that may look like other characters in different fonts
+        // have been omitted.
+        $rString = '';
+        $chars = 'ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz23456789';
+        $charsTotal  = strlen($chars);
+        for ($i = 0; $i < $len; $i++) {
+            $rInt = (integer) mt_rand(0, $charsTotal);
+            $rString .= substr($chars, $rInt, 1);
+        }
+        return $rString;
+    }
+    
+    protected function calculateHash() {
+        return sha1(implode('', $_SESSION['csrf']));
+    }
+
+    public function generateToken() {
+        $_SESSION['csrf'] = array();
```

Comment:
Extract this to the property of this class. in case we need to render two forms in one view, we can customize the key.